docs/src/user_docs/stdlib/math/u64.md

```diff
@@ -0,0 +1,69 @@
+```
+[a_hi, a_lo, ... ]
+```
+
+Procedures which check whether the input values are encoded correctly are designated with `checked` prefix. For example, `checked_add` would fail if any of the top 4 elements on the stack contains a value greater than $2^{32} - 1$. In contrast, `wrapping_add` and `overflowing_add` would not perform these checks, and therefore, if any of the top 4 stack elements is greater than $2^{32} - 1$, the operation will not fail but rather will produce an undefined result. Thus, when using versions of procedures which are not `checked`, it is important to be certain that input values encode valid u64 values.
+
+## Arithmetic operations
+
+| Procedure   | Description   |
+| ----------- | ------------- |
+| overflowing_add |  Performs addition of two unsigned 64 bit integers preserving the overflow.<br /> The input values are assumed to be represented using 32 bit limbs, but this is not checked.<br /> Stack transition looks as follows:<br /> [b_hi, b_lo, a_hi, a_lo, ...] -> [overflowing_flag, c_hi, c_lo, ...], where c = (a + b) % 2^64 |
+| wrapping_add |  Performs addition of two unsigned 64 bit integers discarding the overflow.<br /> The input values are assumed to be represented using 32 bit limbs, but this is not checked.<br /> Stack transition looks as follows:<br /> [b_hi, b_lo, a_hi, a_lo, ...] -> [c_hi, c_lo, ...], where c = (a + b) % 2^64 |
+| checked_add |  Performs addition of two unsigned 64 bit integers, fails when overflowing.<br /> The input values are assumed to be represented using 32 bit limbs, fails if they are not.<br /> Stack transition looks as follows:<br /> [b_hi, b_lo, a_hi, a_lo, ...] -> [c_hi, c_lo, ...], where c = (a + b) % 2^64 |
+| wrapping_sub |  Performs subtraction of two unsigned 64 bit integers discarding the overflow.<br /> The input values are assumed to be represented using 32 bit limbs, but this is not checked.<br /> Stack transition looks as follows:<br /> [b_hi, b_lo, a_hi, a_lo, ...] -> [c_hi, c_lo, ...], where c = (a - b) % 2^64 |
+| checked_sub |  Performs subtraction of two unsigned 64 bit integers, fails when underflowing.<br /> The input values are assumed to be represented using 32 bit limbs, fails if they are not.<br /> Stack transition looks as follows:<br /> [b_hi, b_lo, a_hi, a_lo, ...] -> [c_hi, c_lo, ...], where c = (a - b) % 2^64 |
+| overflowing_sub |  Performs subtraction of two unsigned 64 bit integers preserving the overflow.<br /> The input values are assumed to be represented using 32 bit limbs, but this is not checked.<br /> Stack transition looks as follows:<br /> [b_hi, b_lo, a_hi, a_lo, ...] -> [underflowing_flag, c_hi, c_lo, ...], where c = (a - b) % 2^64 |
```

Comment:
Similar comment to my comment on `overflowing_add` regarding the "underflowing_flag" here

Reply:
Here it is also more like a flag: 1 if there is underflow and 0 otherwise.

Reply:
Yes, you're right. I was thinking about this first in the mul case, and also the description didn't feel consistent because of the different terms used, but yes, the meaning is the same for this case & add, so it's fine